assets/tests/TaskSchemaValidationService.js

```diff
@@ -0,0 +1,225 @@
+          if (this.hasOwnProperty(l) && 
+              this[l] instanceof Function &&
+              !(/functions/i).test(l) &&
+              l.toString().startsWith('verify')) {
+            functions.push(this[l]);
+          }
+        }
+        return functions;
+      },
+      init: function(starterCode, auxiliaryCode) {
+        this._starterCode = starterCode;
+        this._auxiliaryCode = auxiliaryCode;
+      },
+      verifyInstructionsAreNotEmpty: function(task) {
+        var instructions = task.getInstructions();
+        return instructions.length > 0;
+      },
+      verifyInstructionsIsArrayOfStrings: function(task) {
+        var instructions = task.getInstructions();
+        return (angular.isArray(instructions) &&
+          instructions.every(function(instruction) {
+            return angular.isString(instruction);
+          }));
+      },
+      verifyMainFunctionNameIsString: function(task) {
+        var mainFunctionName = task.getMainFunctionName();
+        return (angular.isString(mainFunctionName) &&
+            mainFunctionName.length > 0);
+      },
+      verifyMainFunctionNameAppearsInStarterCode: function(
+        task) {
+        var mainFunctionName = task.getMainFunctionName();
+        return _checkIfFunctionExistsInCode(mainFunctionName, 
+          taskSchemaValidator._starterCode);
+      },
+      verifyInputFunctionNameIsNullOrString: function(task) {
+        var inputFunctionName = task.getInputFunctionName();
+        return (inputFunctionName === null ||
+          angular.isString(inputFunctionName));
+      },
+      verifyInputFunctionNameAppearsInAuxiliaryCodeIfNotNull: function(
+        task) {
+        var inputFunctionName = task.getInputFunctionName();
+        return (inputFunctionName === null ||
+          _checkIfFunctionExistsInClass(
+              inputFunctionName, AUXILIARY_CODE_CLASS_NAME, 
+              taskSchemaValidator._auxiliaryCode));
+      },
+      verifyOutputFunctionNameIsNullOrString: function(task) {
+        var outputFunctionName = task.getOutputFunctionName();
+        return (outputFunctionName === null ||
+          angular.isString(outputFunctionName));
+      },
+      verifyOutputFunctionNameAppearsInAuxiliaryCodeIfNotNull: function(
+        task) {
+        var outputFunctionName = task.getOutputFunctionName();
+        return (outputFunctionName === null ||
+          _checkIfFunctionExistsInClass(
+              outputFunctionName, AUXILIARY_CODE_CLASS_NAME, 
+              taskSchemaValidator._auxiliaryCode));
+      },
+      verifyPrerequisiteSkillsAreArray: function(task) {
+        var prerequisiteSkills = task.getPrerequisiteSkills();
+        return angular.isArray(prerequisiteSkills);
+      },
+      verifyAcquiredSkillsAreArray: function(task) {
+        var acquiredSkills = task.getAcquiredSkills();
+        return angular.isArray(acquiredSkills);
+      },
+      verifyNoDuplicateSkillsInPrompts: function(task) {
```

Comment:
Yes, currently. I imagine you can improve in a skill that's still required for a task.